README.md

```diff
@@ -143,6 +150,32 @@ end
 
 Then follow Devise's general [OmniAuth tutorial](https://github.com/plataformatec/devise/wiki/OmniAuth:-Overview), replacing references to `facebook` with `saml`.
 
+## Single Logout
+
+Single Logout can be Service Provider initiated or Identity Provider initiated.
+The SP initiated flow can be integrated in the `SessionsController#destroy`.
+For this to work its important to preserve the `saml_uid` value before Devise
+clears the session and redirect to the spslo subpath to initiate the single logout.
+Example `destroy` action in `sessions_controller.rb`:
+
+```ruby
+class SessionsController < Devise::SessionsController
+  # ...
+
+  def destroy
+    # Preserve the saml_uid in the session
+    saml_uid = session["saml_uid"]
+    super do
+      session["saml_uid"] = saml_uid
+      if ! SAML_SETTINGS.idp_slo_target_url.nil?
```

Comment:
This looks like a double negative and is bound to cause confusion. You are doing `if not nil` which is the same as `if present?`, which is easier to understand. I know it's just the example in the README, but ti should be easy to read.

Please change it to:

```ruby
if SAML_SETTINGS.idp_slo_target_url.present?
```